tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,282 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/m3x/pool"
+
+	"github.com/m3db/m3db/encoding"
+	"github.com/m3db/m3db/encoding/m3tsz"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/retention"
+	"github.com/m3db/m3db/storage/block"
+	"github.com/m3db/m3db/storage/bootstrap"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper"
+	commitlogsrc "github.com/m3db/m3db/storage/bootstrap/bootstrapper/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3db/x/io"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+	xtime "github.com/m3db/m3x/time"
+)
+
+var (
+	pathPrefixArg           = flag.String("path-prefix", "/var/lib/m3db", "Path prefix - must contain a folder called 'commitlogs'")
+	namespaceArg            = flag.String("namespace", "metrics", "Namespace")
+	blockSizeArg            = flag.Duration("block-size", 10*time.Minute, "Block size")
+	flushSizeArg            = flag.Int("flush-size", 524288, "Flush size of commit log")
+	bootstrapRetentionArg   = flag.Duration("retention", 48*time.Hour, "Retention")
+	shardsCountArg          = flag.Int("shards-count", 8192, "Shards count - set number too bootstrap all shards in range")
+	shardsArg               = flag.String("shards", "", "Shards - set comma separated list of shards")
+	debugListenAddressArg   = flag.String("debug-listen-address", "", "Debug listen address - if set will expose pprof, i.e. ':8080'")
+	currentUnixTimestampArg = flag.Int64("current-unix-timestamp", time.Now().Unix(), "Current unix timestamp (Seconds) - If set will perform the bootstrap as if this was the current time, defaults to current time")
```

Comment:
Instead of specifying current timestamp, it'd be more flexible to allow users to specify a start and an end time.

Also if they don't specify either it'd be nice if it simply tried to verify all commit logs by looking at the very first commit log and very last and work it ot.

Reply:
@robskillington Do I need to look at the commitlog files on disk, or can I just set the start/end times to time.Unix(0, 0) and time.Unix(1<<63-62135596801, 999999999) respectively (https://stackoverflow.com/questions/25065055/what-is-the-maximum-time-time-in-go)